### PR TITLE
Upgrade rust to 1.24

### PR DESCRIPTION
**Pull request by sfackler**

### CODE REVIEW

File: library/rust [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 
 Maintainers: Steven Fackler <sfackler@gmail.com> (@sfackler)
 GitRepo: https://github.com/rust-lang-nursery/docker-rust.git
 
-Tags: 1.23.0-stretch, 1-stretch, 1.23-stretch, stretch, 1.23.0, 1, 1.23, latest
+Tags: 1.24.0-stretch, 1-stretch, 1.24-stretch, stretch, 1.24.0, 1, 1.24, latest
 Architectures: amd64, arm32v7, arm64v8, i386
-GitCommit: e382ab252c238ab77f0b6e45b0705ba94b4e9e1d
-Directory: 1.23.0/stretch
+GitCommit: e840f001dc6f4128b4a3b5add7d03457bab806a9
```
Comment (yosifkit) [inline library/rust]: These commits should probably be [c481589e552336cffcbe97b6e42a7d1d3e1e9a49](https://github.com/rust-lang-nursery/docker-rust/commit/c481589e552336cffcbe97b6e42a7d1d3e1e9a49). :wink:

Comment (sfackler) [reply library/rust]: Oops, not sure what happened there - fixed.

**Comment by yosifkit** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-list b/_bashbrew-list
index 448a678..28524b9 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -1,12 +1,12 @@
 rust:1
 rust:1-jessie
 rust:1-stretch
-rust:1.23
-rust:1.23-jessie
-rust:1.23-stretch
-rust:1.23.0
-rust:1.23.0-jessie
-rust:1.23.0-stretch
+rust:1.24
+rust:1.24-jessie
+rust:1.24-stretch
+rust:1.24.0
+rust:1.24.0-jessie
+rust:1.24.0-stretch
 rust:jessie
 rust:latest
 rust:stretch
diff --git a/rust_jessie/Dockerfile b/rust_jessie/Dockerfile
index c9611a6..7b8833b 100644
--- a/rust_jessie/Dockerfile
+++ b/rust_jessie/Dockerfile
@@ -9,18 +9,18 @@ RUN set -eux; \
 # this "case" statement is generated via "update.sh"
     dpkgArch="$(dpkg --print-architecture)"; \
 	case "${dpkgArch##*-}" in \
-		amd64) rustArch='x86_64-unknown-linux-gnu'; rustupSha256='4b7a67cd971d713e0caef48b5754190aca19192d1863927a005c3432512b12dc' ;; \
-		armhf) rustArch='armv7-unknown-linux-gnueabihf'; rustupSha256='622190c3f478a56563d45f6fbc1fab02d356b631c28a1beba2c3e4c68de3c14c' ;; \
-		arm64) rustArch='aarch64-unknown-linux-gnu'; rustupSha256='a39d7643cdced9ad70a9927bbb0a861b579884f94793881b771d3a0f92c0ddd8' ;; \
-		i386) rustArch='i686-unknown-linux-gnu'; rustupSha256='9e921fce4a2cc1f04095be6d623effdead0aab1261472e6933da9e6030330b90' ;; \
+		amd64) rustArch='x86_64-unknown-linux-gnu'; rustupSha256='c9837990bce0faab4f6f52604311a19bb8d2cde989bea6a7b605c8e526db6f02' ;; \
+		armhf) rustArch='armv7-unknown-linux-gnueabihf'; rustupSha256='297661e121048db3906f8c964999f765b4f6848632c0c2cfb6a1e93d99440732' ;; \
+		arm64) rustArch='aarch64-unknown-linux-gnu'; rustupSha256='a68ac2d400409f485cb22756f0b3217b95449884e1ea6fd9b70522b3c0a929b2' ;; \
+		i386) rustArch='i686-unknown-linux-gnu'; rustupSha256='27e6109c7b537b92a6c2d45ac941d959606ca26ec501d86085d651892a55d849' ;; \
 		*) echo >&2 "unsupported architecture: ${dpkgArch}"; exit 1 ;; \
 	esac; \
     \
-    url="https://static.rust-lang.org/rustup/archive/1.9.0/${rustArch}/rustup-init"; \
+    url="https://static.rust-lang.org/rustup/archive/1.11.0/${rustArch}/rustup-init"; \
     wget "$url"; \
     echo "${rustupSha256} *rustup-init" | sha256sum -c -; \
     chmod +x rustup-init; \
-    ./rustup-init -y --no-modify-path --default-toolchain 1.23.0; \
+    ./rustup-init -y --no-modify-path --default-toolchain 1.24.0; \
     rm rustup-init; \
     chmod -R a+w $RUSTUP_HOME $CARGO_HOME; \
     rustup --version; \
diff --git a/rust_latest/Dockerfile b/rust_latest/Dockerfile
index 744bcee..a5f29f1 100644
--- a/rust_latest/Dockerfile
+++ b/rust_latest/Dockerfile
@@ -9,18 +9,18 @@ RUN set -eux; \
 # this "case" statement is generated via "update.sh"
     dpkgArch="$(dpkg --print-architecture)"; \
 	case "${dpkgArch##*-}" in \
-		amd64) rustArch='x86_64-unknown-linux-gnu'; rustupSha256='4b7a67cd971d713e0caef48b5754190aca19192d1863927a005c3432512b12dc' ;; \
-		armhf) rustArch='armv7-unknown-linux-gnueabihf'; rustupSha256='622190c3f478a56563d45f6fbc1fab02d356b631c28a1beba2c3e4c68de3c14c' ;; \
-		arm64) rustArch='aarch64-unknown-linux-gnu'; rustupSha256='a39d7643cdced9ad70a9927bbb0a861b579884f94793881b771d3a0f92c0ddd8' ;; \
-		i386) rustArch='i686-unknown-linux-gnu'; rustupSha256='9e921fce4a2cc1f04095be6d623effdead0aab1261472e6933da9e6030330b90' ;; \
+		amd64) rustArch='x86_64-unknown-linux-gnu'; rustupSha256='c9837990bce0faab4f6f52604311a19bb8d2cde989bea6a7b605c8e526db6f02' ;; \
+		armhf) rustArch='armv7-unknown-linux-gnueabihf'; rustupSha256='297661e121048db3906f8c964999f765b4f6848632c0c2cfb6a1e93d99440732' ;; \
+		arm64) rustArch='aarch64-unknown-linux-gnu'; rustupSha256='a68ac2d400409f485cb22756f0b3217b95449884e1ea6fd9b70522b3c0a929b2' ;; \
+		i386) rustArch='i686-unknown-linux-gnu'; rustupSha256='27e6109c7b537b92a6c2d45ac941d959606ca26ec501d86085d651892a55d849' ;; \
 		*) echo >&2 "unsupported architecture: ${dpkgArch}"; exit 1 ;; \
 	esac; \
     \
-    url="https://static.rust-lang.org/rustup/archive/1.9.0/${rustArch}/rustup-init"; \
+    url="https://static.rust-lang.org/rustup/archive/1.11.0/${rustArch}/rustup-init"; \
     wget "$url"; \
     echo "${rustupSha256} *rustup-init" | sha256sum -c -; \
     chmod +x rustup-init; \
-    ./rustup-init -y --no-modify-path --default-toolchain 1.23.0; \
+    ./rustup-init -y --no-modify-path --default-toolchain 1.24.0; \
     rm rustup-init; \
     chmod -R a+w $RUSTUP_HOME $CARGO_HOME; \
     rustup --version; \
```

</details>

**Comment by yosifkit** [comment]
Build test of #4016; 194cf2f8; `amd64` (`rust`):

```console

$ bashbrew build rust:1.24.0-stretch
Building bashbrew/cache:a099fffe8c5d685e848e17a80c6b11729116c25afd05686fb13363fec6563501 (rust:1.24.0-stretch)
Tagging rust:1.24.0-stretch
Tagging rust:1-stretch
Tagging rust:1.24-stretch
Tagging rust:stretch
Tagging rust:1.24.0
Tagging rust:1
Tagging rust:1.24
Tagging rust:latest

$ test/run.sh rust:1.24.0-stretch
testing rust:1.24.0-stretch
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed


$ bashbrew build rust:1.24.0-jessie
Building bashbrew/cache:b5d962218a5837fecaa0bc5d82ee6d5eeff93e651c49cb2944d908d1d50a6b02 (rust:1.24.0-jessie)
Tagging rust:1.24.0-jessie
Tagging rust:1-jessie
Tagging rust:1.24-jessie
Tagging rust:jessie

$ test/run.sh rust:1.24.0-jessie
testing rust:1.24.0-jessie
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed

```

**Pull request merged**